mixer/adapter/redisquota/config/config.proto

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 the Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package adapter.redisquota.config;
+
+import "google/protobuf/duration.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "config";
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+// redisquota adapter supports the rate limit quota using either fixed or
+// rolling window algorithm. And it is using Redis as a shared data storage.
+//
+// Example configuration:
+//
+// redisServerUrl: localhost:6379
+// connectionPoolSize: 10
+// quotas:
+//   - name: requestCount.quota.istio-system
+//     maxAmount: 50
+//     validDuration: 60s
+//     bucketDuration: 1s
+//     rateLimitAlgorithm: ROLLING_WINDOW
+//     overrides:
+//       - dimensions:
+//           destination: ratings
+//           source: reviews
+//         maxAmount: 12
+//       - dimensions:
+//           destination: reviews
+//         maxAmount: 5
+message Params {
+  message Override {
+    option (gogoproto.goproto_getters) = true;
+
+    // The specific dimensions for which this override applies.
+    // String representation of instance dimensions is used to check against configured dimensions.
+    // dimensions should not be empty
+    map <string, string> dimensions = 1 [(gogoproto.nullable) = false];
+
+    // The upper limit for this quota override.
+    // This value should be bigger than 0
+    int64 max_amount = 2;
+  }
+
+  // Algorithms for rate-limiting:
+  // FIXED_WINDOW - The fixed window approach can allow 2x peak specified rate, whereas the rolling-window doesn't.
```

Comment:
Could you put the enum value comments above the declaration of the actual value? That way, it'll come out properly in the docs.

Reply:
Done